packages/angular_devkit/build_angular/src/differential-loading/update-webpack-config.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+type WebpackConfig = any; // tslint:disable-line
+type WebpackPlugin = any; // tslint:disable-line
+
+export function updateWebpackConfig(config: WebpackConfig): WebpackConfig {
+
+  const legacyConfig = {
+    ...config,
+    output: buildLegacyOutput(config),
+    plugins: buildLegacyPlugins(config),
+    resolve: buildLegacyResolve(config),
+  };
+
+  const modernConfig = {
+    ...config,
+    output: buildModernOutput(config),
+    plugins: buildModernPlugins(config),
+    entry: buildModernEntry(config),
+    resolve: buildModernResolve(config),
+  };
+
+  const newConfig = [legacyConfig, modernConfig];
+
+  return newConfig;
+}
+
+function buildLegacyOutput(config: WebpackConfig): object {
+  const filename = config.output.filename.replace('[name]', '[name].legacy');
+
+  return { ...config.output, filename };
+}
+
+function buildModernOutput(config: WebpackConfig): object {
+  const filename = config.output.filename.replace('[name]', '[name].modern');
+
+  return { ...config.output, filename };
+}
+
+function buildModernResolve(config: WebpackConfig): object {
+  const modernResolve = { ...config.resolve };
+  modernResolve.mainFields = ['es2015', 'module', 'browser', 'main'];
+
+  return modernResolve;
+}
+
+function buildLegacyResolve(config: WebpackConfig): object {
+  const modernResolve = { ...config.resolve };
+  modernResolve.mainFields = ['module', 'browser', 'main'];
+
+  return modernResolve;
+}
+
+function buildModernEntry(config: WebpackConfig): object {
+  const entry = config.entry;
+  const polyfills: string[] = entry.polyfills;
+
+  if (!polyfills) {
+    return { ...entry };
+  }
+
+  const tweakPolyfills = (file: string) => {
+
+    // TODO: Do we need an own polyfill file for modern browsers? If yes, where is it created?
```

Comment:
when do we plan to fix the TODO? can we figure it out in this PR?

Reply:
Question for reviewers:  Do we need an own polyfill file for modern browsers? If yes, where is it created?